### PR TITLE
Feature/checkout functionality

### PR DESCRIPTION
**Pull request by HIMDanny**
Add checkout functionality for logged in user

### CODE REVIEW

File: client/src/store/slices/orderSlice/orderSlice.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    increaseStep: (state) => {
+      state.currentStep += 1;
+    },
+    decreaseStep: (state) => {
+      if (state.currentStep === 0) {
+        state.currentStep = 0;
+      } else {
+        state.currentStep -= 1;
+      }
+    },
```
Comment (RomanRymarchuk1) [inline client/src/store/slices/orderSlice/orderSlice.js]:
```suggestion
    decreaseStep: (state) => {
      if (state.currentStep >= 1) {
        state.currentStep -= 1;
      } 
    },
```

perhaps this approach would be better

Comment (HIMDanny) [reply client/src/store/slices/orderSlice/orderSlice.js]: done

**Review by RomanRymarchuk1 — APPROVED** [review]

File: client/src/features/checkout/components/CheckoutForm/CheckoutForm.js [hunk 2]
```diff
@@ -1,63 +1,82 @@
...
+} from '../../../../store/slices/orderSlice/orderSlice';
 // Child Forms and model
 import { UserDetailsForm, ShippingAddressForm, PaymentForm, PaymentSuccess, CheckoutSummary } from '..';
 import { initialValues, validationSchema } from '../../data';
 
 const steps = ['User Details', 'Shipping Address', 'Payment Details'];
 
 const CheckoutForm = () => {
-  const [activeStep, setActiveStep] = useState(0);
+  const { currentStep, isLoading } = useSelector((state) => state.order, shallowEqual);
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/features/checkout/components/CheckoutForm/CheckoutForm.js:25]: i guess shallowEqual is only used for complex data types like arrays and objects which is not our case here

Comment (Ahmed-Fahmy-Dev) [reply client/src/features/checkout/components/CheckoutForm/CheckoutForm.js:25]: already discussed

**Review by Ahmed-Fahmy-Dev — APPROVED** [review]

**Pull request merged**